.github/workflows/sdk-codegen-ci.yml

```diff
@@ -0,0 +1,140 @@
+
+  build-aws-sdk-go-v2:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: 'zulu'
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
+
+      - uses: actions/checkout@v1
+      - id: smithy-version
+        uses: ./.github/workflows/checkout-and-build-smithy
+
+      - name: Add `smithyVersion` env variable
+        run: echo "smithyVersion=${{ steps.smithy-version.outputs.smithy-version }}" >> $GITHUB_ENV
+
+      - name: Checkout smithy-typescript repo
+        uses: actions/checkout@v2
+        with:
+          repository: AndrewFossAWS/smithy-go
+          path: smithy-go
+
+      - name: Update smithy version smithy-go
+        run: cd ./smithy-go/codegen && sed -i "1s/.*/smithyVersion=$smithyVersion/" gradle.properties && cat gradle.properties
+
+      - name: Build smithy-go
+        run: cd ./smithy-go/codegen && ./gradlew smithy-ci
+
+      - name: Checkout aws-sdk-go-v2 repo
+        uses: actions/checkout@v2
+        with:
+          repository: AndrewFossAWS/aws-sdk-go-v2
+          path: aws-sdk-go-v2
+
+      - name: Update smithy version in aws-sdk-go-v2
+        run: cd ./aws-sdk-go-v2/codegen && sed -i "1s/.*/smithyVersion=$smithyVersion/" gradle.properties && cat gradle.properties
+
+      - name: Build aws-sdk-js-v3
+        run: cd ./aws-sdk-go-v2/codegen && ./gradlew smithy-ci
+
+  build-aws-sdk-kotlin:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: 'zulu'
+
+      - uses: actions/checkout@v1
+      - id: smithy-version
+        uses: ./.github/workflows/checkout-and-build-smithy
+
+      - name: Add `smithyVersion` env variable
+        run: echo "smithyVersion=${{ steps.smithy-version.outputs.smithy-version }}" >> $GITHUB_ENV
+
+      - name: Checkout smithy-kotlin repo
+        uses: actions/checkout@v2
+        with:
+          repository: AndrewFossAWS/smithy-kotlin
+          path: smithy-kotlin
+
+      - name: Update smithy version in smithy-kotlin
+        run: cd ./smithy-kotlin && sed -i "25s/.*/smithyVersion=$smithyVersion/" gradle.properties && cat gradle.properties
+
+      - name: Build smithy-kotlin
+        run: cd ./smithy-kotlin && ./gradlew smithy-ci
+
+      - name: Checkout aws-sdk-kotlin
+        uses: actions/checkout@v2
+        with:
+          repository: AndrewFossAWS/aws-sdk-kotlin
+          path: aws-sdk-kotlin
+
+      - name: Update smithy version in aws-sdk-kotlin
+        run: cd ./aws-sdk-kotlin && sed -i "13s/.*/smithyVersion=$smithyVersion/" gradle.properties && cat gradle.properties
+
+      - name: Build aws-sdk-kotlin
+        run: cd ./aws-sdk-kotlin && ./gradlew clean build && ./gradlew -p aws-runtime publishToMavenLocal && ./gradlew -p codegen/protocol-tests testAllProtocols
```

Comment:
this can be simplified to `cd ./aws-sdk-kotlin && ./gradlew publishToMavenLocal && ./gradlew testAllProtocols`